contrib/lkt/language/parser.py

```diff
@@ -2579,6 +2579,23 @@ def called_decl():
             ),
             # If not a generic decl, fall back on the initially found decl
             default_val=called_decl
+        )._.match(
+            # Here, if we found a declaration but it's not one of the
+            # statically callable declarations (so FunDecl and TypeDecl), we
+            # return nothing, even though we might be able to statically
+            # determine a declaration, like in the example below::
+            #
+            #    val a : (Int) -> Int = null
+            #    val b : Int = a(12)
+            #
+            # That's because in language semantic terms, the actual called
+            # function is determined dynamically in the above case.
+            #
+            # TODO: Introduce a base class, so we don't have to do this ugly
+            # match?
```

Comment:
Can you clarify? (a base class for what? how woud it help to avoid the match? I don’t understand why it’s ugly, …)

Reply:
It's ugly because if we had a base class we could just do `.cast(StaticCallableDecl)` or sth